### ISSUE TITLE
onDismiss still not working on 5.0.5

### ISSUE THREAD

**Issue by joewang704** [pburtchaell/react-notification#70]
The following code works with the onClick but does not dismiss after 2 seconds. I tried with manually adding in a dismissAfter value, still doesn't seem to work. I tried logging from the function and it just never gets called. Any ideas why?
```javascript
<Notification
  isActive={true}
  message={notification.message}
  action={"Dismiss"}
  onClick={removeNotification}
  onDismiss={removeNotification}
/>
```

**Comment by radicaled** [comment]
I think this is because the code is using `componentWillReceiveProps` in order to set the dismiss timer, however this isn't present on the initial render of a `<Notification />`, which means the timer never gets triggered.

https://facebook.github.io/react/tips/componentWillReceiveProps-not-triggered-after-mounting.html

I've worked around it by copy-pasting the code here into my app, for the time being: https://github.com/pburtchaell/react-notification/blob/master/src/notification.js#L18

**Comment by pburtchaell** [comment]
So this is probably a bug, or at least something we should rethink, but due to the way it is handled, you need to initially mount the component with `isActive` having a value of `false`. Otherwise, the timeout for dismiss will not trigger. 

```js
<Notification
  message={notification.message}
  action={"Dismiss"}
  onClick={removeNotification}
  onDismiss={removeNotification}
/>
```

**Comment by joewang704** [comment]
@pburtchaell Ok so then the problem is probably because you're using componentWillReceiveProps as @radicaled mentioned, which doesn't get called on the initial render, only after you change the value of a prop after rendering, which is what you're mentioning with isActive.

**Comment by pburtchaell** [comment]
Correct, @joewang704. I'll need to change how this is implemented.

**Comment by pburtchaell** [comment]
This is patched in `react-notification@5.0.7`.
